libcxx/include/__utility/small_buffer.h

```diff
@@ -38,8 +39,10 @@
 _LIBCPP_BEGIN_NAMESPACE_STD
 
 template <size_t _BufferSize, size_t _BufferAlignment>
-  requires(_BufferSize > 0 && _BufferAlignment > 0)
 class __small_buffer {
+  static_assert(std::has_single_bit(_BufferAlignment), "Alignment is invalid.");
+  static_assert(_BufferSize >= sizeof(byte*), "Buffer has to be capable of storing a pointer for heap allocations!");
+
```

Comment:
question: why the condition to use small buffer need `is_trivially_move_constructible_v` and `is_trivially_destructible_v`? Can we allow any types that is `no_throw_move_constructible`?